### PR TITLE
CPopulation done, CCopPed and fixes

### PR DESCRIPTION
**Pull request by erorcun**

### CODE REVIEW

File: src/peds/CopPed.cpp [hunk 1]
```diff
@@ -66,7 +70,210 @@ CCopPed::~CCopPed()
...
+
+void
+CCopPed::ScanForCrimes(void)
+{
+	CVehicle *playerVeh = FindPlayerVehicle();
+
+	// Look for car alarms
+	if (playerVeh && playerVeh->IsCar()) {
+		if (playerVeh->IsAlarmOn()) {
+			if ((playerVeh->GetPosition() - GetPosition()).MagnitudeSqr() < 400.0f)
```
Comment (Nick007J) [inline src/peds/CopPed.cpp]: SQR(20.0f)

File: src/peds/Population.cpp [hunk 2]
```diff
@@ -1005,6 +1001,154 @@ CPopulation::TestRoomForDummyObject(CObject *obj)
...
+							255, 255, 255, 255, 4.0f, 40000, 1.0f);
+						break;
+					case 1:
+						CShadows::AddPermanentShadow(SHADOWTYPE_DARK, gpOutline2Tex, &pedPos,
+							0.9f * cos(randAngle), 0.9f * sin(randAngle), 0.9f * sin(randAngle), -0.9f * cos(randAngle),
+							255, 255, 255, 255, 4.0f, 40000, 1.0f);
+						break;
+					case 2:
+						CShadows::AddPermanentShadow(SHADOWTYPE_DARK, gpOutline3Tex, &pedPos,
+							0.9f * cos(randAngle), 0.9f * sin(randAngle), 0.9f * sin(randAngle), -0.9f * cos(randAngle),
```
Comment (ShFil119) [inline src/peds/Population.cpp]: math pointing to wrappers.

File: src/peds/Population.cpp [hunk 3]
```diff
@@ -1005,6 +1001,154 @@ CPopulation::TestRoomForDummyObject(CObject *obj)
...
+	CVector colCentre = dummy->GetBoundCentre();
+
+	int minX = CWorld::GetSectorIndexX(dummy->GetPosition().x - colRadius);
+	if (minX < 0) minX = 0;
+	int minY = CWorld::GetSectorIndexY(dummy->GetPosition().y - colRadius);
+	if (minY < 0) minY = 0;
+	int maxX = CWorld::GetSectorIndexX(dummy->GetPosition().x + colRadius);
+	if (maxX > NUMSECTORS_X) maxX = NUMSECTORS_X;
+	int maxY = CWorld::GetSectorIndexY(dummy->GetPosition().y + colRadius);
+	if (maxY > NUMSECTORS_Y) maxY = NUMSECTORS_Y;
```
Comment (ShFil119) [inline src/peds/Population.cpp]: min/max?

File: src/peds/CopPed.cpp [hunk 4]
```diff
@@ -66,7 +70,210 @@ CCopPed::~CCopPed()
...
+	m_nLastPedState = PED_NONE;
+	bIsRunning = false;
+	bNotAllowedToDuck = false;
+	bKindaStayInSamePlace = false;
+	m_bZoneDisabledButClose = false;
+	m_bZoneDisabled = false;
+	ClearObjective();
+	if (IsPedInControl()) {
+		if (!m_pMyVehicle || wanted->m_nWantedLevel != 0)  {
+			if (m_pMyVehicle && (m_pMyVehicle->GetPosition() - GetPosition()).MagnitudeSqr() < 25.0f) {
```
Comment (ShFil119) [inline src/peds/CopPed.cpp]: `SQR(5.f)`

File: src/peds/Population.cpp (hunk 3 above)
Comment (erorcun) [reply src/peds/Population.cpp]: aap does the same, and there are many places that I may need to change if I attempt that, so I don't wanna do that :/

**Pull request merged**